tabular/src/autogluon/tabular/models/widedeep_nn/metrics.py

```diff
@@ -0,0 +1,90 @@
+import logging
+from inspect import isclass
+
+from autogluon.core.constants import BINARY, REGRESSION, MULTICLASS
+
+logger = logging.getLogger(__name__)
+
+
+def get_metrics_map(num_classes):
+    from pytorch_widedeep.metrics import Accuracy, R2Score, F1Score
+    import torchmetrics as tm
+    num_classes = 2 if num_classes is None else num_classes
+    metrics_map = {
+        # Regression
+        'root_mean_squared_error': tm.MeanSquaredError(squared=False),
+        'mean_squared_error': tm.MeanSquaredError(),
+        'mean_absolute_error': tm.MeanAbsoluteError(),
+        'r2': R2Score,
+        # Not supported: 'median_absolute_error': None,
+
+        # Classification
+        'accuracy': Accuracy,
+
+        'f1': F1Score,
+        'f1_macro': tm.F1Score(average='macro', num_classes=num_classes),
+        'f1_micro': tm.F1Score(average='micro', num_classes=num_classes),
+        'f1_weighted': tm.F1Score(average='weighted', num_classes=num_classes),
+
+        'roc_auc': tm.AUROC(num_classes=num_classes),
+
+        'precision': tm.Precision(num_classes=num_classes),
+        'precision_macro': tm.Precision(average='macro', num_classes=num_classes),
+        'precision_micro': tm.Precision(average='micro', num_classes=num_classes),
+        'precision_weighted': tm.Precision(average='weighted', num_classes=num_classes),
+
+        'recall': tm.Recall(num_classes=num_classes),
+        'recall_macro': tm.Recall(average='macro', num_classes=num_classes),
+        'recall_micro': tm.Recall(average='micro', num_classes=num_classes),
+        'recall_weighted': tm.Recall(average='weighted', num_classes=num_classes),
+        'log_loss': None,
+
+        # Not supported: 'pinball_loss': None
+        # Not supported: pac_score
+    }
+    return metrics_map
+
+
+def get_nn_metric(problem_type, stopping_metric, num_classes):
+    metrics_map = get_metrics_map(num_classes)
+
+    # Unsupported metrics will be replaced by defaults for a given problem type
+    objective_func_name = stopping_metric.name
```

Comment:
metric != objective_func, why not `metric_name`?

Reply:
good suggestion - updated